src/Comp.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import {useState} from 'react';
+function Comp()
+{
+    const[count, setCount] = useState(0);
+    const[comment, setComment] = useState([]);
+
+    let myRef = React.createRef();
+
+    let handler = () => {
+        let currentCount = count;
+        currentCount++;
+        setCount(currentCount);
+    }
+
+    let addComment = () => {
+        let commentValue = myRef.current.value;
+        let comments = [...comment, commentValue];
+        setComment(comments);
+        myRef.current.value='';
+    }
+    
+    return(
+        <>
+        <h1>Hi</h1>
+        
+        <div>
+            <input ref={myRef}></input>
+        </div>
+        <div>
+            <button onClick={addComment}>Submit</button>
+        </div>
+        <div>
+            <ul>
+                {comment.map((item,index) => <li key={index.toString()}>{item}</li>)}
```

Comment:
key не должен быть index массива, это плохая практика и несет за собой поледствия
https://reactjs.org/docs/lists-and-keys.html

> We don’t recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state. Check out Robin Pokorny’s article for an in-depth explanation on the negative impacts of using an index as a key. If you choose not to assign an explicit key to list items then React will default to using indexes as keys.